File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectModel/src/main/java/ro/mpp2025/javaprojectui/Participant.java
//package ro.mpp2025.javaprojectui;
//
//public class Participant extends ro.mpp2025.javaprojectui.Entity<Integer> {
//
//    private String nume;
//    private String cnp;
//
//    public Participant (String nume, String cnp) {
//        this.nume = nume;
//        this.cnp = cnp;
//    }
//
//    public String getNume() {
//        return nume;
//    }
//
//    public String getCnp() {
//        return cnp;
//    }
//
//    @Override
//    public boolean equals(Object obj) {
//        return false;
//    }
//
//    @Override
//    public int hashCode() {
//        return 0;
//    }
//
//    @Override
//    public String toString() {
//        return "Participant{" +
//                "nume='" + nume + '\'' +
//                ", cnp='" + cnp + '\'' +
//                '}';
//    }
//}

package ro.mpp2025.javaprojectui;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

@jakarta.persistence.Entity
@Table(name = "participanti")
public class Participant extends ro.mpp2025.javaprojectui.Entity<Integer> {

    @NotNull
    @Column(name = "nume")
    private String nume;

    @NotNull
    @Column(name = "cnp")
    private String cnp;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Override
    public Integer getId() {
        return super.getId();
    }

    @Override
    public void setId(Integer id) {
        super.setId(id);
    }

    public Participant() {}

    public Participant(String nume, String cnp) {
        this.nume = nume;
        this.cnp = cnp;
    }

    public String getNume() {
        return nume;
    }

    public String getCnp() {
        return cnp;
    }

    public void setNume(@NotNull String nume) {
        this.nume = nume;
    }

    public void setCnp(@NotNull String cnp) {
        this.cnp = cnp;
    }

    @Override
    public boolean equals(Object obj) {
        return false;
    }

    @Override
    public int hashCode() {
        return 0;
    }

    @Override
    public String toString() {
        return "Participant{" +
                "nume='" + nume + '\'' +
                ", cnp='" + cnp + '\'' +
                '}';
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Model/Participant.cs
using System.Text.Json.Serialization;

namespace Model;
[Serializable]
public class Participant : Entity<int>
{
    private string _nume;
    private string _cnp;
        
    public Participant(){}
    [JsonConstructor]
    public Participant(string nume, string cnp)
    {
        this._nume = nume;
        this._cnp = cnp;
    }

    public override string ToString()
    {
        return $"{nameof(_nume)}: {_nume}, {nameof(_cnp)}: {_cnp}, {nameof(Nume)}: {Nume}, {nameof(Cnp)}: {Cnp}";
    }

    public string Nume
    {
        get { return _nume; }
        set { _nume = value; }
    }
        
    public string Cnp
    {
        get { return _cnp; }
        set { _cnp = value; }
    }
        
    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;
            
        Participant that = (Participant)obj;
        return Equals(_nume, that._nume) && Equals(_cnp, that._cnp);
    }
        
    public override int GetHashCode()
    {
        return HashCode.Combine(_nume, _cnp);
    }
}